### PR TITLE
skipped test_ser before rebooting a DUT

### PR DESCRIPTION
**Pull request by antonptashnik**
### Description of PR

Summary: skipped test_ser before rebooting a DUT
Fixes # (issue)

platform_tests/broadcom/test_ser test is skipped for platforms other than Broadcom. The issue is that the skip condition executes too late allowing test preparation code to reboot a DUT. Fixed
### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012

#### What is the motivation for this PR?
Reduce test exec time for platforms the test is skipped for
#### How did you do it?
Moved test skip to tests_mark_conditions, so the skip condition will be executed earlier any test preparation code

### CODE REVIEW

**Comment by antonptashnik** [comment]
@ZhaohuiS @wangxin please review

File: tests/common/plugins/conditional_mark/tests_mark_conditions.yaml [hunk 1]
```diff
@@ -753,3 +753,12 @@ vxlan/test_vxlan_ecmp.py:
...
       - "(is_multi_asic==True) or (platform not in ['x86_64-mlnx_msn4600c-r0', 'x86_64-8102_64h_o-r0'])"
+
+#######################################
+#####         test_ser            #####
+#######################################
+platform_tests/broadcom/test_ser.py:
+  skip:
+    reason: "test_ser is only supported for Broadcom"
+    conditions:
+      - "asic_type not in ['broadcom']"
```
Comment (wangxin) [inline tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:764]: Can you move this skip condition into another file `tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml` and ensure alphabetic order?

Comment (antonptashnik) [reply tests/common/plugins/conditional_mark/tests_mark_conditions.yaml]: addressed, please re-review

**Comment by antonptashnik** [comment]
pre-commit finds the issue in the fragment below that is not touched by this PR
```
snmp/test_snmp_queue.py:
  skip:
    reason: "Interfaces not present on supervisor node"
    conditions:
      - "is_supervisor"
  skip:
    reason: "M0 topo does not support test_snmp_queue"
    conditions:
      - "topo_name in ['m0']"
```

**Comment by antonptashnik** [comment]
@wangxin @ZhaohuiS please review

**Comment by ZhaohuiS** [comment]
@lipxu Could you please review it?Thanks.

File: tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml [hunk 2]
```diff
@@ -581,6 +581,19 @@ platform_tests/broadcom:
     conditions:
       - "'marvell' in asic_type"
 
+platform_tests/broadcom/test_ser.py:
```
Comment (lipxu) [inline tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml:584]: test_ser should be the only case in test_ser.py, may we combine these two rules?
like conditions_logical_operator: 'OR'

Comment (antonptashnik) [reply tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml:584]: you're right! I combined 2 skips under a single file section

**Comment by antonptashnik** [comment]
the issue reported by pre-commit is caused by specific structure of .yaml test skip files which allow several `skip` reasons to be specified within the same section, like below:
```
platform_tests/broadcom/test_ser.py:
  skip:
    reason: "platform does not support test_ser"
    conditions:
      - "platform in ['x86_64-cel_e1031-r0']"
      - https://github.com/sonic-net/sonic-mgmt/issues/6218
  skip:
    reason: "test_ser is only supported for Broadcom"
    conditions:
      - "asic_type not in ['broadcom']"
```

File: tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml [hunk 3]
```diff
@@ -581,6 +581,17 @@ platform_tests/broadcom:
...
+  skip:
+    reason: "platform does not support test_ser"
+    conditions:
+      - "platform in ['x86_64-cel_e1031-r0']"
+      - https://github.com/sonic-net/sonic-mgmt/issues/6218
+  skip:
```
Comment (lipxu) [inline tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml:590]: Not sure whether two skips in one rule work, may we try using 'OR' to combine these two skips?
for example:
platform_tests/broadcom/test_ser.py::test_ser:
  skip:
    reason: "platform does not support test_ser"
    conditions_logical_operator: 'OR'
    conditions:
      - "platform in ['x86_64-cel_e1031-r0'] and https://github.com/sonic-net/sonic-mgmt/issues/6218"
      - "asic_type not in ['broadcom']"

Comment (antonptashnik) [reply tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml]: thanks, I applied your proposal

**Review by lipxu — APPROVED** [review]

**Comment by antonptashnik** [comment]
the issue reported by `pre-commit-check` is not related to scope this PR addresses and should be fixed by another PR

**Comment by antonptashnik** [comment]
> the issue reported by `pre-commit-check` is not related to scope this PR addresses and should be fixed by another PR

addressed in https://github.com/sonic-net/sonic-mgmt/pull/6972

**Comment by antonptashnik** [comment]
@ZhaohuiS I made the requested changes. Please merge if possible, thanks !

**Pull request merged**